src/components/Dropdown/index.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useMemo } from 'react';
+import { CSSTransition } from 'react-transition-group';
+import classNames from 'classnames';
+
+import useClickOutside from '../../hooks/useClickOutside';
+
+import Input, { IInput } from '../Input';
+import Elevation from '../Elevation';
+import Button from '../Button';
+import Icon from '../Icons';
+
+import './index.scss';
+
+const TEST_ID = '@fv/dropdown';
+
+export interface IDropdown extends Omit<IInput, 'mask' | 'type' | 'sideContent'> {
+  isOpen?: boolean;
+  closeOnBlur?: boolean;
+  closeOnClickOutside?: boolean;
+  inputClassName?: 'string';
+  onToggle?: (isOpen: boolean) => void;
+  onClickOutside?: React.MouseEventHandler<HTMLButtonElement>;
+  onIconClick?: React.MouseEventHandler<HTMLButtonElement>;
+}
+
+export const DropDown: React.FC<IDropdown> = (props) => {
+  const {
+    isOpen,
+    name,
+    children,
+    className,
+    disabled,
+    onToggle,
+    onFocus,
+    onBlur,
+    onIconClick,
+    onClickOutside,
+    inputClassName,
+    closeOnBlur,
+    closeOnClickOutside,
+    ...rest
+  } = props;
+
+  const controlled = useMemo(() => isOpen !== undefined && isOpen !== null, [isOpen]);
+  const [isOpenInternal, setOpenInternal] = useState(false);
+
+  const toggleOpenInternal = (isOpen: boolean) => {
+    setOpenInternal(isOpen);
+    onToggle && onToggle(isOpen);
```

Comment:
we can do optional check onToggle?.

Reply:
This is something I'd like to have a discussion if we want an eslint rule to enforce one way or the other - this was consistent with some other files, while I saw that Alex did it with optional check in newer PR's. Not leaning one way or the other, would be nice to have this agreement across the team